crates/jstzd/src/task/octez_node.rs

```diff
@@ -0,0 +1,264 @@
+use super::Task;
+use anyhow::Result;
+use async_dropper::AsyncDrop;
+use async_trait::async_trait;
+use std::{fs::File, path::PathBuf, sync::Arc};
+use tokio::{runtime::Handle, sync::RwLock};
+
+#[cfg(test)]
+use tests::{Child, InnerOctezNode};
+
+#[cfg(not(test))]
+use octez::OctezNode as InnerOctezNode;
+#[cfg(not(test))]
+use std::process::Child;
+
+pub struct OctezNodeConfig {
+    binary_path: PathBuf,
+    data_dir: PathBuf,
+    network: String,
+    http_endpoint: String,
+    rpc_endpoint: String,
+    num_connections: u32,
+    log_file: PathBuf,
+    options: Vec<String>,
+}
+
+impl OctezNodeConfig {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        binary_path: &str,
+        data_dir: &str,
+        network: &str,
+        http_endpoint: &str,
+        rpc_endpoint: &str,
+        num_connections: u32,
+        log_file: &str,
+        options: &[&str],
+    ) -> Self {
+        OctezNodeConfig {
+            binary_path: PathBuf::from(binary_path),
+            data_dir: PathBuf::from(data_dir),
+            network: network.to_owned(),
+            http_endpoint: http_endpoint.to_owned(),
+            rpc_endpoint: rpc_endpoint.to_owned(),
+            num_connections,
+            log_file: PathBuf::from(log_file),
+            options: options
+                .iter()
+                .map(|v| (*v).to_owned())
+                .collect::<Vec<String>>(),
+        }
+    }
+}
+
+#[derive(Default)]
+struct ChildWrapper {
```

Comment:
Isn't this identical to the `AsyncDropper<T>` from `async_dropper_simple`?

Reply:
Yes. I've replaced this with `AsyncDropper`.

Reply:
Current implementation looks good to me. Resolving.